### PR TITLE
fix: improve Django path normalization

### LINKED ISSUE #227: problems normalizing paths causes test failures in django-oscar
Running the tests in [django-oscar](https://github.com/django-oscar/django-oscar) with v1.14.0 of appmap-python shows 4 failures:
```
FAILED tests/unit/catalogue/test_views_reviews.py::TestAddVoteView::test_voting_on_product_review_redirect_on_public_product - re.error: unbalanced parenthesis at position 71
FAILED tests/unit/catalogue/test_views_reviews.py::TestAddVoteView::test_voting_on_product_review_returns_404_on_non_public_product - re.error: unbalanced parenthesis at position 71
FAILED tests/unit/catalogue/test_views_reviews.py::TestProductReviewDetail::test_retrieving_product_review_returns_200_on_public_product - RuntimeError: No match for /catalogue/dumu03fbu03a5-title_12/reviews/70/ found with resolver <URLResolver 'tests._site.urls' (None:None) '^/'>, regex ^catalogue/\(\...
FAILED tests/unit/catalogue/test_views_reviews.py::TestProductReviewDetail::test_retrieving_product_review_returns_404_on_non_public_product - RuntimeError: No match for /catalogue/dumu03fbu03a5-title_13/reviews/71/ found with resolver <URLResolver 'tests._site.urls' (None:None) '^/'>, regex ^catalogue/\(\...
```

One of the failure stack traces:
```
ERROR    django.request:log.py:224 Internal Server Error: /catalogue/dumu03fbu03a5-title_13/reviews/71/
Traceback (most recent call last):
  File "/Users/ajp/src/django-oscar/django-oscar/.direnv/python-3.10.8/lib/python3.10/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/Users/ajp/src/django-oscar/django-oscar/.direnv/python-3.10.8/lib/python3.10/site-packages/appmap/django.py", line 236, in __call__
    rec, start, call_event_id = self.before_request_hook(request, request.path_info)
  File "/Users/ajp/src/django-oscar/django-oscar/.direnv/python-3.10.8/lib/python3.10/site-packages/_appmap/web_framework.py", line 170, in before_request_hook
    start, call_event_id = self.before_request_main(rec, request)
  File "/Users/ajp/src/django-oscar/django-oscar/.direnv/python-3.10.8/lib/python3.10/site-packages/appmap/django.py", line 270, in before_request_main
    normalized_path_info = normalize_path_info(req.path_info, resolved)
  File "/Users/ajp/src/django-oscar/django-oscar/.direnv/python-3.10.8/lib/python3.10/site-packages/appmap/django.py", line 195, in normalize_path_info
    regex, match = get_resolved_match(path_info, resolver)
  File "/Users/ajp/src/django-oscar/django-oscar/.direnv/python-3.10.8/lib/python3.10/site-packages/appmap/django.py", line 181, in get_resolved_match
    raise RuntimeError(
RuntimeError: No match for /catalogue/dumu03fbu03a5-title_13/reviews/71/ found with resolver <URLResolver 'tests._site.urls' (None:None) '^/'>, regex ^catalogue/\(\?P(?P<product_slug>[^/]+)\[\\w\-\]\*\)_\(\?P(?P<product_pk>[^/]+)\\d\+\)/reviews/(?P<pk>[0-9]+)/
```
`

### PR DESCRIPTION
**Pull request by apotterri**
Fix the way path normalization is done for Django http_server_requests. Get rid of the use of private functions (which didn't work in all cases), stick to the public API of ResolverMatch.

Fixes #227

### CODE REVIEW

**Comment by brikelly** [comment]
@dividedmind please review

File: pylintrc [hunk 1]
```diff
@@ -1,33 +1,77 @@
...
+# Files or directories to be skipped. They should be base names, not paths.
 ignore=CVS
 
-# Add files or directories matching the regex patterns to the blacklist. The
-# regex matches against base names, not paths.
+# Add files or directories matching the regular expressions patterns to the
+# ignore-list. The regex matches against paths and can be in Posix or Windows
+# format. Because '\\' represents the directory delimiter on Windows systems,
+# it can't be used as an escape character.
+ignore-paths=^_appmap\\test\\data|^_appmap/test/data
```
Comment (dividedmind) [inline pylintrc:55]: Is this regex actually correct? Seems like you need to parenthesize the two parts. (Also, the comment suggests the setting could be an array, maybe that'd make it easier.)

File: pylintrc [hunk 2]
```diff
@@ -42,6 +86,13 @@ load-plugins=
 # Pickle collected data for later comparisons.
 persistent=yes
 
+# Minimum Python version to use for version dependent checks. Will default to
+# the version used to run pylint.
+py-version=3.10
```
Comment (dividedmind) [inline pylintrc:91]: Should this match the python version requirement from the project file?

File: pylintrc [hunk 3]
```diff
@@ -42,6 +86,13 @@ load-plugins=
...
+
+# Discover python modules and packages in the file system subtree.
+recursive=no
```
Comment (dividedmind) [inline pylintrc:94]: Is this correct?

File: pylintrc [hunk 4]
```diff
@@ -39,7 +39,7 @@ extension-pkg-whitelist=
 fail-on=forgotten-debug-statement
 
 # Specify a score threshold under which the program will exit with error.
-fail-under=9.59
+fail-under=9.74
```
Comment (dividedmind) [inline pylintrc:42]: :+1:

**Review by dividedmind — APPROVED** [review]

**Comment by apotterri** [comment]
I don't know the answer to your questions about `pylintrc`, and the pylint doc stinks. :(

As far as I can tell, the values for all three of these options were generated when I reran `pylint --generate-rcfile`. So, at the very least, I would expect that the value for `ignore-paths` is correct.

I commented out all three, so the default values would be used. That didn't change the final score, so I'm inclined to leave them the way they are.

**Pull request merged**